go/v1beta1/storage/embeddedstore.go

```diff
@@ -0,0 +1,418 @@
+// Copyright 2019 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/boltdb/bolt"
+	"github.com/golang/protobuf/proto"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/grafeas/grafeas/go/errors"
+	"github.com/grafeas/grafeas/go/name"
+	pb "github.com/grafeas/grafeas/proto/v1beta1/grafeas_go_proto"
+	prpb "github.com/grafeas/grafeas/proto/v1beta1/project_go_proto"
+	"golang.org/x/net/context"
+	fieldmaskpb "google.golang.org/genproto/protobuf/field_mask"
+	"google.golang.org/grpc/codes"
+	"log"
+	"os"
+	"path/filepath"
+)
+
+const (
+	bucketOccurrences = "occurrences"
+	bucketProjects    = "projects"
+	bucketNotes       = "notes"
+	bucketOperations  = "operations"
+)
+
+var (
+	errKeyExists = fmt.Errorf("key exists")
+	errNoKey     = fmt.Errorf("key missing")
+)
+
+type EmbeddedStoreConfig struct {
+	Path string `yaml:"path"` // Path is the folder path to storage files
+}
+
+// embeddedStore is a storage solution for Grafeas based on boltdb
+type embeddedStore struct {
```

Comment:
Why is this a private type? I get that the members should be private, but since we're exposing the constructor to end users, we probably want to expose the underlying type as well.

Reply:
This was just copy/paste from the previous implementation, but I agree that it should be public type. Updated.